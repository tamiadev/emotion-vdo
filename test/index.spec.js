/** @jsx vdo */
const vdo = require('vdo');
const styled = require('../src/index');

test('render a component with a class name', () => {
	const Cmpnt = styled('div')`
		color: red;
	`;
	const result = <Cmpnt id="pizza" />;
	expect(result.toString()).toMatchSnapshot();
});

test('omit non-HTML props', () => {
	const Cmpnt = styled('div')`
		color: red;
	`;
	const result = <Cmpnt id="pizza" coffee={42} />;
	expect(result.toString()).toMatchSnapshot();
});

test('render a component with children', () => {
	const Cmpnt = styled('div')`
		color: red;
	`;
	const result = (
		<Cmpnt>
			<h1>Hello</h1>
		</Cmpnt>
	);
	expect(result.toString()).toMatchSnapshot();
});

test('withComponent changes component', () => {
	const Cmpnt = styled('div')`
		color: red;
	`;
	const Cmpnt2 = Cmpnt.withComponent('span');
	const result = <Cmpnt2 />;
	expect(result.toString()).toMatchSnapshot();
});
