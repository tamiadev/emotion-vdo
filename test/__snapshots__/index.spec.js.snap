// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omit non-HTML props 1`] = `"<div id=\\"pizza\\" className=\\"css-17pzvdr\\"></div>"`;

exports[`render a component with a class name 1`] = `"<div id=\\"pizza\\" className=\\"css-17pzvdr\\"></div>"`;

exports[`render a component with children 1`] = `"<div className=\\"css-17pzvdr\\"><h1>Hello</h1></div>"`;

exports[`withComponent changes component 1`] = `"<span className=\\"css-17pzvdr\\"></span>"`;
